scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -812,6 +813,15 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def asIterableSideInput: SideInput[Iterable[T]] =
     new IterableSideInput[T](this.applyInternal(View.asIterable()))
 
+  def asSetSingletonSideInput(implicit coder: Coder[T]): SideInput[Set[T]] =
+    self
+      .aggregate(m.Set[T]())(_ += _, _ ++= _) // Use mutable sets during aggregation
```

Comment:
I wonder if `.aggregate(Set.newBuilder[T]())(_ += _,  _ ++= _.result)` is better since that saves a the `O(n)` `_.toSet`?

Reply:
I tried this out:
```
.aggregate(Set.newBuilder[T])(
        (b: mutable.Builder[T, Set[T]], t: T) => b += t,
        (b: mutable.Builder[T, Set[T]], c: mutable.Builder[T, Set[T]]) => b ++= c.result()
      ) // Use mutable builder during aggregation
    .map(_.result()) // materialize the final builder
```
We still need a call to the `_.result` which will materialize it to an immutable `Set`

The other bigger problem is that the `mutable.Builder[T, Set[T]]` would not have a `Coder[mutable.Builder[T, Set[T]]]` and it is very hard to write a Coder without using Kryo as the Builder is invariant in `T` or materialize the builder every time we want to serialize it.